<?xml version="1.0" encoding="utf-8"?>
<launch>

    <arg name="metric_map_file" />
    <arg name="topo_map_file" />
    <arg name="param_file" />

    <!-- Internal visualization level - for debugging only -->
    <arg name="vis_level" default="0"/>
    <arg name="gdb" default="false"/>

    <node name="map_server" pkg="map_server" type="map_server" args="'$(arg metric_map_file)'">
        <remap from="/map" to="/metric_map" />
    </node>

    <node name="topo_mapping" pkg="infobot_topo_mapping" type="topo_mapping" output="screen" required="true"
          launch-prefix="gdb -ex run --args" if="$(arg gdb)">
        <param name="visualization_level" type="int" value="$(arg vis_level)"/>
        <rosparam command="load" file="$(arg param_file)"/>
        <remap from="~metric_map" to="/metric_map"/>
        <remap from="~topo_map_in" to="/topo_map"/>
        <remap from="~topo_map_out" to="/topo_map"/>
    </node>

    <node name="topo_mapping" pkg="infobot_topo_mapping" type="topo_mapping" output="screen" required="true"
          unless="$(arg gdb)">
        <param name="visualization_level" type="int" value="$(arg vis_level)"/>
        <rosparam command="load" file="$(arg param_file)"/>
        <remap from="~metric_map" to="/metric_map"/>
        <remap from="~topo_map_in" to="/topo_map"/>
        <remap from="~topo_map_out" to="/topo_map"/>
    </node>

    <node pkg="infobot_topo_server" type="topo_saver" name="topo_saver"
          output="screen" args="-f $(arg topo_map_file)">
        <remap from="~topo_map" to="/topo_map"/>
    </node>

</launch>
